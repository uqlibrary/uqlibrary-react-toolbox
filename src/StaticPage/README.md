# StaticPage

component for a simple text page with title and help icon

## Props

- `title: PropTypes.string.isRequired`
- `text: PropTypes.string.isRequired`
- `help: PropTypes.object`, eg {title: 'About', text: 'help text..', buttonLabel:'Close'}

## Usage

**./App.js**
```jsx
import React from 'react';
...

import {StaticPage} from 'uqlibrary-react-toolbox';
        
const App = () => (
  <div>
    <StaticPage 
      title='Contact us' 
      text='contact details here...' 
      help={title: 'About', text: 'help text..', buttonLabel:'Close'} 
       />
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```

