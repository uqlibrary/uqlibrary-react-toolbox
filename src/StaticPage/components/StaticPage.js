import React from 'react';
import PropTypes from 'prop-types';

import {Card, CardText, CardHeader} from 'material-ui/Card';
import {HelpIcon} from '../../index';

export default function StaticPage({title, text, help}) {
    return (

        <div className="layout-fill">

            <h1 className="page-title display-1">{title ? title : 'This is the page title'}</h1>

            <Card className="layout-card">
                <CardHeader className="card-header">
                    <div className="columns is-gapless">

                        <div className="column">
                            <h2 className="headline">{title ? title : 'This is the card title'}</h2>
                        </div>

                        <div className="column is-narrow is-helpicon">
                            {help && (
                                <HelpIcon
                                    title={help.title}
                                    text={help.text}
                                    buttonLabel={help.button}
                                />
                            )}
                        </div>
                    </div>
                </CardHeader>

                <CardText className="body-1">
                    <br />
                    <div>
                        {text ? text : 'This is the default card content'}
                    </div>
                </CardText>

            </Card>
        </div>

    );
}

StaticPage.propTypes = {
    title: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    help: PropTypes.object
};

StaticPage.defaultProps = {};
