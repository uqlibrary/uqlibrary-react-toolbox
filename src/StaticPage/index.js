export {default as StaticPage} from './components/StaticPage';
