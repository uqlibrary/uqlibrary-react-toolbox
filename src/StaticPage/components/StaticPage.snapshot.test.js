jest.dontMock('./StaticPage');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import StaticPage from './StaticPage';

function setup(title, text, help) {
    const props = {title, text, help};
    return shallow(<StaticPage {...props} />);
}

describe('StaticPage snapshots tests', () => {
    it('renders StaticPage with title and text', () => {
        const wrapper = setup('about', 'this is about page');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders StaticPage with title, text and help button', () => {
        const wrapper = setup('about', 'this is about page', {title: 'help', text: 'help text'});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});