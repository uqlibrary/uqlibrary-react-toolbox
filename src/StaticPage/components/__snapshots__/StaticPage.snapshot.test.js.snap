// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticPage snapshots tests renders StaticPage with title and text 1`] = `
<div
  className="layout-fill"
>
  <h1
    className="page-title display-1"
  >
    about
  </h1>
  <Card
    className="layout-card"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardText
      className="body-1"
    >
      <br />
      this is about page
    </CardText>
  </Card>
</div>
`;

exports[`StaticPage snapshots tests renders StaticPage with title, text and help button 1`] = `
<div
  className="layout-fill"
>
  <h1
    className="page-title display-1"
  >
    about
    <Connect(HelpIcon)
      inline={true}
      text="help text"
      title="help"
    />
  </h1>
  <Card
    className="layout-card"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardText
      className="body-1"
    >
      <br />
      this is about page
    </CardText>
  </Card>
</div>
`;
